source/uri-options/tests/tls-options.yml

```diff
@@ -0,0 +1,64 @@
+tests:
+    -
+        description: "Valid required tls options are parsed correctly"
+        uri: "mongodb://example.com/?tls=true&tlsCAFile=ca.pem&tlsClientCertFile=cert.pem&tlsClientKeyPassword=hunter2"
+        valid: true
+        warning: false
+        hosts: ~
+        auth:
+            username: ~
+            password: ~
+            db: ~
+        options:
+            tls: "true"
+            tlsCAFile: "ca.pem"
+            tlsClientCertFile: "cert.pem"
+            tlsClientKeyPassword: "hunter2"
+    -
+        description: "tlsAllowInvalidCertificates is parsed correctly"
```

Comment:
Should there be error tests for these options as well? Looking back, most other test files asserted valid parsing for all related options in the first test and then included several tests that testing individual parse failures.

Reply:
Given that these all accept arbitrary strings, what would an example of an invalid option be? Due to an earlier request, PEM files for the valid names were added to accommodate drivers that immediately check for the existence of the files, but I'd prefer not to mandate that drivers have to check the filesystem during URI parsing.

Reply:
Does `tlsAllowInvalidCertificates` accept an arbitrary string? The tls path options certainly do, but this is a boolean option. I expect drivers could check for errors if a value such as "maybe" were passed for a boolean option (e.g. `tlsAllowInvalidCertificates`, `retryWrites`).

Reply:
I'll add tests to assert that non-boolean values for `tlsAllowInvalidCertificates`, ``tlsAllowInvalidHostnames`, and `tlsInsecure` cause warnings.

Reply:
Implemented